### PR TITLE
[FIX] pairwise_combine: compute size only once

### PR DESCRIPTION
**Pull request by marehr**

### CODE REVIEW

File: include/seqan3/range/views/pairwise_combine.hpp [hunk 1]
```diff
@@ -219,7 +219,8 @@ class pairwise_combine_view : public std::ranges::view_interface<pairwise_combin
         requires std::ranges::sized_range<underlying_range_type>
     //!\endcond
     {
-        return (std::ranges::size(u_range) * (std::ranges::size(u_range) - 1) / 2);
+        auto size = std::ranges::size(u_range);
```
Comment (rrahn) [inline include/seqan3/range/views/pairwise_combine.hpp:222]: Why is this necessary? std::ranges::size requires (semantically) constant time. Is there anything more to it, than just reducing one additional size call?
```suggestion
        auto const size = std::ranges::size(u_range);
```

**Review by Irallia — APPROVED** [review]
LGFM, thx

File: include/seqan3/range/views/pairwise_combine.hpp (hunk 1 above)
Comment (marehr) [reply include/seqan3/range/views/pairwise_combine.hpp]: I can also drop the PR. I found it easier to read `(size * (size - 1)) / 2` and you are correct that size is constant and I at first thought it wasn't, but this change will remove that doubt.

File: include/seqan3/range/views/pairwise_combine.hpp [hunk 2]
```diff
@@ -219,7 +219,8 @@ class pairwise_combine_view : public std::ranges::view_interface<pairwise_combin
         requires std::ranges::sized_range<underlying_range_type>
     //!\endcond
     {
-        return (std::ranges::size(u_range) * (std::ranges::size(u_range) - 1) / 2);
+        auto const size = std::ranges::size(u_range);
```
Comment (smehringer) [inline include/seqan3/range/views/pairwise_combine.hpp:222]: Since `size()` is constant. Does this PR really improve the performance? (Just asking, I don't care for either version evenif it does not)

**Review by smehringer — APPROVED** [review]
Isn't `size()` constant anyway?

File: include/seqan3/range/views/pairwise_combine.hpp (hunk 2 above)
Comment (marehr) [reply include/seqan3/range/views/pairwise_combine.hpp:222]: Nop; it shouldn't improve performance since it is constant time. It's more like a style guide, [FIX] might be the wrong tag for this.

Comment (smehringer) [reply include/seqan3/range/views/pairwise_combine.hpp:222]: Alright.
